vm/src/stdlib/itertools.rs

```diff
@@ -736,6 +737,143 @@ impl PyItertoolsTee {
     }
 }
 
+#[pyclass]
+#[derive(Debug)]
+struct PyIterToolsProduct {
+    pools: Vec<Vec<PyObjectRef>>,
+    idxs: RefCell<Vec<usize>>,
+    cur: RefCell<usize>,
+    sizes: Vec<usize>,
+    stop: RefCell<bool>,
+}
+
+impl PyValue for PyIterToolsProduct {
+    fn class(vm: &VirtualMachine) -> PyClassRef {
+        vm.class("itertools", "product")
+    }
+}
+
+#[derive(FromArgs)]
+struct ProductArgs {
+    #[pyarg(keyword_only, optional = true)]
```

Comment:
This can be an `OptionalArg<usize>` to avoid having to check for an overflow yourself.

Reply:
Can we use `OptionalArg<usize>` directly into `product::tp_new`?